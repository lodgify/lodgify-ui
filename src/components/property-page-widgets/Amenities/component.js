import React from 'react';
import PropTypes from 'prop-types';

import { VIEW_MORE } from 'utils/default-strings';
import { Grid } from 'layout/Grid';
import { GridRow } from 'layout/GridRow';
import { GridColumn } from 'layout/GridColumn';
import { Heading } from 'typography/Heading';

import { getDefaultItems } from './utils/getDefaultItems';
import { hasExtraItems } from './utils/hasExtraItems';
import { getCategoryMarkup } from './utils/getCategoryMarkup';
import { getExtraItemsMarkup } from './utils/getExtraItemsMarkup';
import { getExtraItems } from './utils/getExtraItems';

/**
 * The standard widget for displaying the amenities of a property.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  amenities,
  hasExtraItemsInModal,
  headingText,
  isStacked,
  modalTriggerText,
}) => (
  <Grid className="is-amenities" columns={isStacked ? 1 : 3}>
    <GridRow>
      {headingText && (
        <GridColumn width={12}>
          <Heading>{headingText}</Heading>
        </GridColumn>
      )}
      {getDefaultItems(amenities, isStacked).map((amenity, index) =>
        getCategoryMarkup(amenity, index, isStacked)
      )}
      {hasExtraItems(amenities, isStacked) &&
        getExtraItemsMarkup(
          hasExtraItemsInModal,
          modalTriggerText,
          getExtraItems(amenities, isStacked)
        )}
    </GridRow>
  </Grid>
);

Component.displayName = 'Amenities';

Component.defaultProps = {
  hasExtraItemsInModal: false,
  headingText: null,
  isStacked: false,
  modalTriggerText: VIEW_MORE,
};

Component.propTypes = {
  /** The amenity categories. */
  amenities: PropTypes.arrayOf(
    PropTypes.shape({
      /** The name of the icon to display for the category.
       * [See here for the full list of valid icon names](https://github.com/lodgify/lodgify-ui/blob/master/src/components/elements/Icon/constants.js)
       */
      iconName: PropTypes.string.isRequired,
      /** The list of amenities */
      items: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
      /** The category name */
      name: PropTypes.string.isRequired,
    })
  ).isRequired,
  /** Are the extra items displayed in a modal. */
  hasExtraItemsInModal: PropTypes.bool,
  /** The text to display as a heading at the top of the amenities. */
  headingText: PropTypes.string,
  /** Are the amenities displayed stacked on top of one another */
  isStacked: PropTypes.bool,
  /** The text for the modal trigger */
  modalTriggerText: PropTypes.string,
};
