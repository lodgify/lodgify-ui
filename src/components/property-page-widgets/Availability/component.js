import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { CalendarMonth, CalendarDay } from 'react-dates';
import { BLOCKED_MODIFIER } from 'react-dates/constants';
import moment from 'moment';
import { Card } from 'semantic-ui-react';
import { size } from 'lodash';

import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { Dropdown } from 'inputs/Dropdown';
import { Grid } from 'layout/Grid';
import { GridColumn } from 'layout/GridColumn';
import { GridRow } from 'layout/GridRow';
import { Heading } from 'typography/Heading';
import { Icon, ICON_NAMES } from 'elements/Icon';
import { Paragraph } from 'typography/Paragraph';
import { withResponsive } from 'utils/with-responsive';
import {
  AVAILABILITY,
  NEXT,
  PREVIOUS,
  PROPERTIES,
  UNAVAILABLE,
  VIEW_AVAILABILITY_FOR,
} from 'utils/default-strings';

import { BLOCKED_DAY_CLASS } from './constants';
import { isDayBlockedOrBeforeCurrentDate } from './utils/isDayBlockedOrBeforeCurrentDate';
import { getMonthsToDisplay } from './utils/getMonthsToDisplay';
import { getNextStartDate } from './utils/getNextStartDate';
import { getPreviousStartDate } from './utils/getPreviousStartDate';
import { renderMonthHeader } from './utils/renderMonthHeader';

import 'react-dates/initialize';

/**
 * The standard widget for displaying the availability of a room.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
class Component extends PureComponent {
  state = {
    startDate: moment(),
  };

  componentDidMount = () => moment.locale(this.props.localeCode);

  handleClickNextMonth = () =>
    this.setState({
      startDate: getNextStartDate(
        this.state.startDate,
        this.props.isUserOnMobile
      ),
    });

  handleClickPreviousMonth = () =>
    this.setState({
      startDate: getPreviousStartDate(
        this.state.startDate,
        this.props.isUserOnMobile
      ),
    });

  renderCalendarDay = props => {
    const modifiers = new Set([]);
    const { getIsDayBlocked } = this.props;

    if (isDayBlockedOrBeforeCurrentDate(props.day, getIsDayBlocked)) {
      modifiers.add(BLOCKED_DAY_CLASS);
      modifiers.add(BLOCKED_MODIFIER);
    }

    return <CalendarDay {...props} modifiers={modifiers} />;
  };

  reloadCalendarOnRoomSelection = (evt, value) => {
    this.props.onChangeRoomDropdown(evt, value);
    this.forceUpdate();
  };

  render() {
    const { startDate } = this.state;
    const {
      headingText,
      isUserOnMobile,
      legendLabel,
      nextMonthsButtonLabel,
      previousMonthsButtonLabel,
      propertyDropdownLabel,
      propertyDropdownPlaceholderLabel,
      roomOptionsWithImages,
    } = this.props;

    return (
      <div>
        <Heading size="small">{headingText}</Heading>
        <Grid>
          <GridRow>
            {size(roomOptionsWithImages) > 0 && (
              <GridColumn
                computer={7}
                mobile={12}
                tablet={6}
                verticalAlignContent="middle"
              >
                <Grid>
                  <GridColumn
                    computer={3}
                    mobile={5}
                    tablet={12}
                    verticalAlignContent="middle"
                  >
                    <Paragraph size="tiny" weight="heavy">
                      {propertyDropdownLabel}
                    </Paragraph>
                  </GridColumn>
                  <GridColumn computer={7} mobile={7} tablet={12}>
                    <Dropdown
                      icon={ICON_NAMES.MAP_PIN}
                      label={propertyDropdownPlaceholderLabel}
                      onChange={this.reloadCalendarOnRoomSelection}
                      options={roomOptionsWithImages}
                    />
                  </GridColumn>
                </Grid>
              </GridColumn>
            )}
            <GridColumn
              floated="right"
              only="tablet computer"
              textAlign="right"
              verticalAlignContent="middle"
              width={5}
            >
              <Icon
                color="light grey"
                isLabelLeft
                labelText={legendLabel}
                name={ICON_NAMES.SQUARE}
              />
            </GridColumn>
          </GridRow>
        </Grid>
        <Card fluid>
          <Grid padded>
            <GridRow>
              {getMonthsToDisplay(startDate, isUserOnMobile).map(
                (month, index) => (
                  <GridColumn
                    className="availability-calendar-wrapper"
                    computer={6}
                    key={buildKeyFromStrings('month-column', index)}
                    mobile={12}
                    tablet={6}
                  >
                    <CalendarMonth
                      isVisible
                      month={month}
                      renderCalendarDay={this.renderCalendarDay}
                      renderMonthElement={renderMonthHeader}
                    />
                  </GridColumn>
                )
              )}
            </GridRow>
            <GridRow>
              <GridColumn width={6}>
                <Icon
                  labelText={previousMonthsButtonLabel}
                  name={ICON_NAMES.ARROW_LEFT}
                  onClick={this.handleClickPreviousMonth}
                />
              </GridColumn>
              <GridColumn textAlign="right" width={6}>
                <Icon
                  isLabelLeft
                  labelText={nextMonthsButtonLabel}
                  name={ICON_NAMES.ARROW_RIGHT}
                  onClick={this.handleClickNextMonth}
                />
              </GridColumn>
            </GridRow>
          </Grid>
        </Card>
        <Grid>
          <GridColumn only="mobile" textAlign="right" width={12}>
            <GridColumn>
              <Icon
                color="light grey"
                isLabelLeft
                labelText={legendLabel}
                name={ICON_NAMES.SQUARE}
              />
            </GridColumn>
          </GridColumn>
        </Grid>
      </div>
    );
  }
}

export const ComponentWithResponsive = withResponsive(Component);

Component.displayName = 'Availability';

Component.defaultProps = {
  getIsDayBlocked: () => false,
  headingText: AVAILABILITY,
  legendLabel: UNAVAILABLE,
  localeCode: 'en',
  nextMonthsButtonLabel: NEXT,
  onChangeRoomDropdown: Function.prototype,
  previousMonthsButtonLabel: PREVIOUS,
  propertyDropdownLabel: VIEW_AVAILABILITY_FOR,
  propertyDropdownPlaceholderLabel: PROPERTIES,
  roomOptionsWithImages: [],
};

Component.propTypes = {
  /**
   * A function that is passed to determine if the current date should be blocked, the function is passed a moment object
   * @param {Moment} momentObj
   * @returns{boolean}
   */
  getIsDayBlocked: PropTypes.func,
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  isUserOnMobile: PropTypes.bool.isRequired,
  /** The label for the legend. */
  legendLabel: PropTypes.string,
  /** The ISO 639-1 locale code which changes the format and language of days of the week and the months of the year. */
  localeCode: PropTypes.string,
  /** The label for the next months button. */
  nextMonthsButtonLabel: PropTypes.string,
  /**
   * A function that triggered when a room is changed in the room dropdown
   * @param {Event} event
   * @param {String} currentRoomValue
   */
  onChangeRoomDropdown: PropTypes.func,
  /** The label for the previous months button. */
  previousMonthsButtonLabel: PropTypes.string,
  /** The label that goes next to the property dropdown. */
  propertyDropdownLabel: PropTypes.string,
  /** The label that goes inside the dropdown when there are no options. */
  propertyDropdownPlaceholderLabel: PropTypes.string,
  /** The options which the user can select for the room field. */
  roomOptionsWithImages: PropTypes.PropTypes.arrayOf(
    PropTypes.shape({
      /** A list of one or more strings separated by commas indicating a set of source sizes for the image. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      imageSizes: PropTypes.string,
      /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use for the image. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      imageSrcSet: PropTypes.string,
      /** The source url of the image to display with the option. */
      imageUrl: PropTypes.string,
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ),
};
