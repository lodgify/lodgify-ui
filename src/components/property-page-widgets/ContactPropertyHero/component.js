import React, { Fragment } from 'react';
import { string, bool, shape, number, oneOfType, arrayOf } from 'prop-types';

import { FullBleed } from 'media/FullBleed';
import { ContactHeader } from 'collections/ContactHeader';
import { FlexContainer } from 'layout/FlexContainer';
import { toUpper } from 'utils/to-upper';
import { Button } from 'elements/Button';
import { ICON_NAMES } from 'elements/Icon';
import { HorizontalGutters } from 'layout/HorizontalGutters';
import { Gallery } from 'media/Gallery';
import { Divider } from 'elements/Divider';
import { size } from 'utils/size';
import { VIEW_MORE_PICTURES } from 'utils/default-strings';

import { BOTTOM_OFFSET } from './constants';

/**
 * A hero displays a contact header with a gallery.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  backgroundImageHeight,
  backgroundImageSizes,
  backgroundImageSrcSet,
  backgroundImageUrl,
  backgroundImageWidth,
  bottomOffset,
  className,
  currencyOptions,
  currencyValue,
  headingHref,
  headingText,
  languageOptions,
  languageValue,
  navigationMenuItems,
  phoneNumber,
  placeholderBackgroundImageUrl,
  primaryButtonText,
  galleryImages,
  secondaryButtonText,
  currencyNoResultsText,
}) => {
  return (
    <FullBleed
      bottomOffset={bottomOffset}
      className={className}
      hasGradient
      imageHeight={backgroundImageHeight}
      imageUrl={backgroundImageUrl}
      imageWidth={backgroundImageWidth}
      placeholderImageUrl={placeholderBackgroundImageUrl}
      sizes={backgroundImageSizes}
      srcSet={backgroundImageSrcSet}
    >
      <ContactHeader
        className={className}
        currencyNoResultsText={currencyNoResultsText}
        currencyOptions={currencyOptions}
        currencyValue={currencyValue}
        headingHref={headingHref}
        headingText={headingText}
        isBackgroundFilled
        languageOptions={languageOptions}
        languageValue={languageValue}
        navigationMenuItems={navigationMenuItems}
        phoneNumber={phoneNumber}
        primaryButtonText={primaryButtonText}
      />
      {size(galleryImages) > 1 ? (
        <Fragment>
          <FlexContainer alignItems="flex-end">
            <HorizontalGutters>
              <Gallery
                images={galleryImages}
                trigger={
                  <Button
                    icon={ICON_NAMES.PLACEHOLDER}
                    isCompact
                    isPositionedRight
                    isSecondary
                  >
                    {toUpper(secondaryButtonText)}
                  </Button>
                }
              />
            </HorizontalGutters>
          </FlexContainer>
          <Divider />
        </Fragment>
      ) : null}
    </FullBleed>
  );
};

Component.displayName = 'ContactPropertyHero';

Component.defaultProps = {
  backgroundImageHeight: undefined,
  backgroundImageSizes: undefined,
  backgroundImageSrcSet: undefined,
  backgroundImageWidth: undefined,
  bottomOffset: BOTTOM_OFFSET,
  className: null,
  currencyNoResultsText: undefined,
  currencyOptions: [],
  currencyValue: null,
  headingHref: '',
  headingText: '',
  languageOptions: [],
  languageValue: null,
  phoneNumber: '',
  placeholderBackgroundImageUrl: null,
  galleryImages: [],
  secondaryButtonText: VIEW_MORE_PICTURES,
  primaryButtonText: '',
};

Component.propTypes = {
  /** The natural height of the background image. */
  backgroundImageHeight: oneOfType([string, number]),
  /** A list of one or more strings separated by commas indicating a set of source sizes for the background image. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  backgroundImageSizes: string,
  /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use for the background image. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  backgroundImageSrcSet: string,
  /** The source url of the hero's background image. */
  backgroundImageUrl: string.isRequired,
  /** The natural width of the background image. */
  backgroundImageWidth: oneOfType([string, number]),
  /** Reduce the height of the Hero with an offset, supports CSS dimensions. */
  bottomOffset: string,
  /** The custom classes. */
  className: string,
  /** The text to display when no results are returned from the currency dropdown. */
  currencyNoResultsText: string,
  /** The options which the user can select for the currency dropdown. */
  currencyOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the currency dropdown. */
  currencyValue: oneOfType([bool, number, string]),
  /** The images to display in the gallery modal. */
  galleryImages: arrayOf(
    shape({
      /** A description of the image to show above the slideshow when the image is showing. */
      descriptionText: string,
      /** A list of one or more strings separated by commas indicating a set of source sizes. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      sizes: string,
      /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      srcSet: string,
      /** Title of the image to show when hovering over it on desktop browsers. */
      title: string,
      /** URL pointing to the image to display. */
      url: string.isRequired,
    })
  ),
  /** The link for the heading. */
  headingHref: string,
  /** The text for the heading. */
  headingText: string,
  /** The options which the user can select for the language dropdown. */
  languageOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the language dropdown. */
  languageValue: oneOfType([bool, number, string]),
  /** The items for a user to navigate the site. */
  navigationMenuItems: arrayOf(
    shape({
      /** The href url for an item which is a link. */
      href: string,
      /** Sub navigation items to display as a [`<Submenu />`](#submenu) under an item. */
      subItems: arrayOf(
        shape({
          href: string.isRequired,
          target: string,
          text: string.isRequired,
        })
      ),
      /** Specifies where to display the linked navigation items URL. See [MDN docs `<a />` for more](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target). */
      target: string,
      /** The visible text for an item. */
      text: string.isRequired,
    })
  ).isRequired,
  /** The phone number to display. */
  phoneNumber: string,
  /** The ISO 639-1 locale code which changes the format and language of days of the week and the months of the year in the search bars date range picker. */
  placeholderBackgroundImageUrl: string,
  /** The text to display for the primary button. */
  primaryButtonText: string,
  /** The text to display on the secondary button at the bottom of the hero. */
  secondaryButtonText: string,
};
