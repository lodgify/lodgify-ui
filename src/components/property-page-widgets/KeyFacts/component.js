import React from 'react';
import PropTypes from 'prop-types';
import { Label } from 'semantic-ui-react';

import { KEY_FACTS } from 'utils/default-strings';
import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { Heading } from 'typography/Heading';
import { Grid } from 'layout/Grid';
import { GridColumn } from 'layout/GridColumn';
import { IconCard } from 'elements/IconCard';

/**
 * The standard widget for displaying key facts about a property.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ keyFacts, headingText }) => (
  <Grid>
    <GridColumn width={12}>
      <Heading>{headingText}</Heading>
    </GridColumn>
    <GridColumn width={12}>
      <Label.Group>
        {keyFacts.map(({ iconName, isDisabled, label }, index) => (
          <IconCard
            isDisabled={isDisabled}
            isFilled
            key={buildKeyFromStrings(label, index)}
            label={label}
            name={iconName}
          />
        ))}
      </Label.Group>
    </GridColumn>
  </Grid>
);

Component.displayName = 'KeyFacts';

Component.defaultProps = {
  headingText: KEY_FACTS,
};

Component.propTypes = {
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** The key facts to display as icon cards. */
  keyFacts: PropTypes.arrayOf(
    PropTypes.shape({
      /**
       * The name of the icon to display.
       * [See here for the full list of valid icon names](https://github.com/lodgify/lodgify-ui/blob/master/src/components/elements/Icon/constants.js)
       */
      iconName: PropTypes.string.isRequired,
      /** Is the key fact disabled. */
      isDisabled: PropTypes.bool,
      /** A visible label to display for the key fact. */
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
};
