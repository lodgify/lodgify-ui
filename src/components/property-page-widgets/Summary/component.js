import React from 'react';
import PropTypes from 'prop-types';
import { Segment } from 'semantic-ui-react';

import { withResponsive } from 'utils/with-responsive';

import { getHeadingMarkup } from './utils/getHeadingMarkup';
import { getNightPriceAndRatingMarkup } from './utils/getNightPriceAndRatingMarkup';
import { getNightPriceRatingAndLocationMarkup } from './utils/getNightPriceRatingAndLocationMarkup';

/**
 * The standard widget for displaying the summary details of a property.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
const Component = ({
  locationName,
  nightPrice,
  propertyName,
  ratingNumber,
  isUserOnMobile,
}) => (
  <Segment.Group compact>
    {!isUserOnMobile && getHeadingMarkup(propertyName)}
    <Segment.Group horizontal={!isUserOnMobile}>
      {isUserOnMobile
        ? getNightPriceAndRatingMarkup(ratingNumber, nightPrice)
        : getNightPriceRatingAndLocationMarkup(
            ratingNumber,
            nightPrice,
            locationName
          )}
    </Segment.Group>
  </Segment.Group>
);

Component.displayName = 'Summary';

Component.propTypes = {
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  isUserOnMobile: PropTypes.bool.isRequired,
  /** The name of the location of the property. */
  locationName: PropTypes.string.isRequired,
  /** The price per night of the property, with currency symbol. */
  nightPrice: PropTypes.string.isRequired,
  /** The name of the property. */
  propertyName: PropTypes.string.isRequired,
  /** The numeral rating for the property, out of 5 */
  ratingNumber: PropTypes.number.isRequired,
};

export const ComponentWithResponsive = withResponsive(Component);
