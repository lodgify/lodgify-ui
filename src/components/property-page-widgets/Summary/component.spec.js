import React from 'react';
import { mount } from 'enzyme';
import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';

import { Component as Summary } from './component';

const props = {
  locationName: 'Catania',
  nightPrice: '$280',
  propertyName: 'The Cat House',
  ratingNumber: 4.8,
};

const getSummary = extraProps => mount(<Summary {...props} {...extraProps} />);

describe('<Summary/>', () => {
  describe('by default', () => {
    it('should render the right structure', () => {
      const actual = getSummary();

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `props.areOnlyNightPriceAndRatingDisplayed` is true', () => {
    it('should render the right children', () => {
      const actual = getSummary({ areOnlyNightPriceAndRatingDisplayed: true });

      expect(actual).toMatchSnapshot();
    });
  });

  it('should have `displayName` Summary', () => {
    expectComponentToHaveDisplayName(Summary, 'Summary');
  });
});
