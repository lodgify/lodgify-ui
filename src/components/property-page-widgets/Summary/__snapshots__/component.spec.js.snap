// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary/> by default should render the right structure 1`] = `
<Summary
  areOnlyNightPriceAndRatingDisplayed={false}
  locationName="Catania"
  nightPrice="$280"
  propertyName="The Cat House"
  ratingNumber={4.8}
>
  <SegmentGroup
    className="is-summary"
    compact={true}
  >
    <div
      className="ui compact segments is-summary"
    >
      <Segment>
        <div
          className="ui segment"
        >
          <Heading
            className={null}
            isColorInverted={false}
            size="medium"
            textAlign={null}
          >
            <Header
              as="h3"
              className={null}
              inverted={false}
              textAlign={null}
            >
              <h3
                className="ui header"
              >
                The Cat House
              </h3>
            </Header>
          </Heading>
        </div>
      </Segment>
      <SegmentGroup
        className="horizontal-wrap"
        horizontal={true}
      >
        <div
          className="ui horizontal segments horizontal-wrap"
        >
          <Segment
            className="is-location-name"
            title="Catania"
          >
            <div
              className="ui segment is-location-name"
              title="Catania"
            >
              Catania
              <Icon
                color="yellow"
                hasBorder={false}
                isButton={false}
                isCircular={false}
                isColorInverted={false}
                isDisabled={false}
                isLabelLeft={false}
                labelText={null}
                labelWeight={null}
                name="map pin"
                path={null}
                size="small"
              >
                <i
                  className="icon yellow small"
                >
                  <svg
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12,2.72c-3.17,0-5.74,3-5.74,6.69S12,21.28,12,21.28s5.74-8.17,5.74-11.87S15.17,2.72,12,2.72Zm0,9.74a4,4,0,1,1,4-4A4,4,0,0,1,12,12.46Z"
                      fill="currentColor"
                    />
                  </svg>
                </i>
              </Icon>
            </div>
          </Segment>
          <Segment
            className="is-rating"
          >
            <div
              className="ui segment is-rating"
            >
              <Rating
                iconSize="tiny"
                isShowingNumeral={true}
                ratingNumber={4.8}
              >
                4.8
                <Rating
                  clearable="auto"
                  disabled={true}
                  maxRating={5}
                  rating={5}
                  size="tiny"
                >
                  <div
                    className="ui tiny disabled rating"
                    onMouseLeave={[Function]}
                    role="radiogroup"
                  >
                    <RatingIcon
                      active={true}
                      aria-checked={false}
                      aria-posinset={1}
                      aria-setsize={5}
                      as="i"
                      index={0}
                      key="0"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      selected={false}
                    >
                      <i
                        aria-checked={false}
                        aria-posinset={1}
                        aria-setsize={5}
                        className="active icon"
                        onClick={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        role="radio"
                        tabIndex={0}
                      />
                    </RatingIcon>
                    <RatingIcon
                      active={true}
                      aria-checked={false}
                      aria-posinset={2}
                      aria-setsize={5}
                      as="i"
                      index={1}
                      key="1"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      selected={false}
                    >
                      <i
                        aria-checked={false}
                        aria-posinset={2}
                        aria-setsize={5}
                        className="active icon"
                        onClick={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        role="radio"
                        tabIndex={0}
                      />
                    </RatingIcon>
                    <RatingIcon
                      active={true}
                      aria-checked={false}
                      aria-posinset={3}
                      aria-setsize={5}
                      as="i"
                      index={2}
                      key="2"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      selected={false}
                    >
                      <i
                        aria-checked={false}
                        aria-posinset={3}
                        aria-setsize={5}
                        className="active icon"
                        onClick={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        role="radio"
                        tabIndex={0}
                      />
                    </RatingIcon>
                    <RatingIcon
                      active={true}
                      aria-checked={false}
                      aria-posinset={4}
                      aria-setsize={5}
                      as="i"
                      index={3}
                      key="3"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      selected={false}
                    >
                      <i
                        aria-checked={false}
                        aria-posinset={4}
                        aria-setsize={5}
                        className="active icon"
                        onClick={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        role="radio"
                        tabIndex={0}
                      />
                    </RatingIcon>
                    <RatingIcon
                      active={true}
                      aria-checked={true}
                      aria-posinset={5}
                      aria-setsize={5}
                      as="i"
                      index={4}
                      key="4"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      selected={false}
                    >
                      <i
                        aria-checked={true}
                        aria-posinset={5}
                        aria-setsize={5}
                        className="active icon"
                        onClick={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        role="radio"
                        tabIndex={0}
                      />
                    </RatingIcon>
                  </div>
                </Rating>
              </Rating>
            </div>
          </Segment>
          <Segment>
            <div
              className="ui segment"
            >
              <span>
                from 
                <Heading
                  className={null}
                  isColorInverted={false}
                  size="small"
                  textAlign={null}
                >
                  <Header
                    as="h4"
                    className={null}
                    inverted={false}
                    textAlign={null}
                  >
                    <h4
                      className="ui header"
                    >
                      $280
                    </h4>
                  </Header>
                </Heading>
                /night
              </span>
            </div>
          </Segment>
        </div>
      </SegmentGroup>
    </div>
  </SegmentGroup>
</Summary>
`;

exports[`<Summary/> if \`props.areOnlyNightPriceAndRatingDisplayed\` is true should render the right children 1`] = `
<Summary
  areOnlyNightPriceAndRatingDisplayed={true}
  locationName="Catania"
  nightPrice="$280"
  propertyName="The Cat House"
  ratingNumber={4.8}
>
  <SegmentGroup
    className="is-summary"
    compact={true}
  >
    <div
      className="ui compact segments is-summary"
    >
      <SegmentGroup
        className="horizontal-wrap"
        horizontal={false}
      >
        <div
          className="ui segments horizontal-wrap"
        >
          <Segment>
            <div
              className="ui segment"
            >
              <span>
                from 
                <Heading
                  className={null}
                  isColorInverted={false}
                  size="small"
                  textAlign={null}
                >
                  <Header
                    as="h4"
                    className={null}
                    inverted={false}
                    textAlign={null}
                  >
                    <h4
                      className="ui header"
                    >
                      $280
                    </h4>
                  </Header>
                </Heading>
                /night
              </span>
            </div>
          </Segment>
          <Segment
            className="is-rating"
          >
            <div
              className="ui segment is-rating"
            >
              <Rating
                iconSize="tiny"
                isShowingNumeral={true}
                ratingNumber={4.8}
              >
                4.8
                <Rating
                  clearable="auto"
                  disabled={true}
                  maxRating={5}
                  rating={5}
                  size="tiny"
                >
                  <div
                    className="ui tiny disabled rating"
                    onMouseLeave={[Function]}
                    role="radiogroup"
                  >
                    <RatingIcon
                      active={true}
                      aria-checked={false}
                      aria-posinset={1}
                      aria-setsize={5}
                      as="i"
                      index={0}
                      key="0"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      selected={false}
                    >
                      <i
                        aria-checked={false}
                        aria-posinset={1}
                        aria-setsize={5}
                        className="active icon"
                        onClick={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        role="radio"
                        tabIndex={0}
                      />
                    </RatingIcon>
                    <RatingIcon
                      active={true}
                      aria-checked={false}
                      aria-posinset={2}
                      aria-setsize={5}
                      as="i"
                      index={1}
                      key="1"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      selected={false}
                    >
                      <i
                        aria-checked={false}
                        aria-posinset={2}
                        aria-setsize={5}
                        className="active icon"
                        onClick={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        role="radio"
                        tabIndex={0}
                      />
                    </RatingIcon>
                    <RatingIcon
                      active={true}
                      aria-checked={false}
                      aria-posinset={3}
                      aria-setsize={5}
                      as="i"
                      index={2}
                      key="2"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      selected={false}
                    >
                      <i
                        aria-checked={false}
                        aria-posinset={3}
                        aria-setsize={5}
                        className="active icon"
                        onClick={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        role="radio"
                        tabIndex={0}
                      />
                    </RatingIcon>
                    <RatingIcon
                      active={true}
                      aria-checked={false}
                      aria-posinset={4}
                      aria-setsize={5}
                      as="i"
                      index={3}
                      key="3"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      selected={false}
                    >
                      <i
                        aria-checked={false}
                        aria-posinset={4}
                        aria-setsize={5}
                        className="active icon"
                        onClick={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        role="radio"
                        tabIndex={0}
                      />
                    </RatingIcon>
                    <RatingIcon
                      active={true}
                      aria-checked={true}
                      aria-posinset={5}
                      aria-setsize={5}
                      as="i"
                      index={4}
                      key="4"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      selected={false}
                    >
                      <i
                        aria-checked={true}
                        aria-posinset={5}
                        aria-setsize={5}
                        className="active icon"
                        onClick={[Function]}
                        onKeyUp={[Function]}
                        onMouseEnter={[Function]}
                        role="radio"
                        tabIndex={0}
                      />
                    </RatingIcon>
                  </div>
                </Rating>
              </Rating>
            </div>
          </Segment>
        </div>
      </SegmentGroup>
    </div>
  </SegmentGroup>
</Summary>
`;
